packages/@n8n/nodes-langchain/nodes/chains/InformationExtractor/InformationExtractor.node.ts

```diff
@@ -0,0 +1,307 @@
+										value: 'date',
+									},
+									{
+										name: 'Number',
+										value: 'number',
+									},
+									{
+										name: 'String',
+										value: 'string',
+									},
+								],
+								default: 'string',
+							},
+							{
+								displayName: 'Description',
+								name: 'description',
+								type: 'string',
+								default: '',
+								description: 'Describe your attribute',
+								placeholder: 'Add description for the attribute',
+								required: true,
+							},
+							{
+								displayName: 'Required',
+								name: 'required',
+								type: 'boolean',
+								default: false,
+								description: 'Whether attribute is required',
+								required: true,
+							},
+						],
+					},
+				],
+			},
+			{
+				displayName: 'Options',
+				name: 'options',
+				type: 'collection',
+				default: {},
+				placeholder: 'Add Option',
+				options: [
+					{
+						displayName: 'System Prompt Template',
+						name: 'systemPromptTemplate',
+						type: 'string',
+						default: SYSTEM_PROMPT_TEMPLATE,
+						description: 'String to use directly as the system prompt template',
+						typeOptions: {
+							rows: 6,
+						},
+					},
+				],
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+
+		const llm = (await this.getInputConnectionData(
+			NodeConnectionType.AiLanguageModel,
+			0,
+		)) as BaseLanguageModel;
+
+		const options = this.getNodeParameter('options', 0, {}) as {
+			systemPromptTemplate?: string;
+		};
+
+		const schemaType = this.getNodeParameter('schemaType', 0, '') as
+			| 'fromAttributes'
+			| 'fromJson'
+			| 'manual';
+
+		let parser: OutputFixingParser<object>;
+
+		if (schemaType === 'fromAttributes') {
+			const attributes = this.getNodeParameter(
+				'attributes.attributes',
+				0,
+				[],
+			) as AttributeDefinition[];
+
+			if (attributes.length === 0) {
+				throw new NodeOperationError(this.getNode(), 'At least one attribute must be specified');
+			}
+
+			parser = OutputFixingParser.fromLLM(
+				llm,
+				StructuredOutputParser.fromZodSchema(makeZodSchemaFromAttributes(attributes)),
+			);
+		} else {
+			let jsonSchema: JSONSchema7;
+
+			if (schemaType === 'fromJson') {
+				const jsonExample = this.getNodeParameter('jsonSchemaExample', 0, '') as string;
+				jsonSchema = generateSchema(jsonExample);
+			} else {
+				const inputSchema = this.getNodeParameter('inputSchema', 0, '') as string;
+				jsonSchema = jsonParse<JSONSchema7>(inputSchema);
+			}
+
+			const zodSchemaSandbox = getSandboxWithZod(this, jsonSchema, 0);
+			const zodSchema = (await zodSchemaSandbox.runCode()) as z.ZodSchema<object>;
+
+			parser = OutputFixingParser.fromLLM(llm, StructuredOutputParser.fromZodSchema(zodSchema));
+		}
+
+		const systemPromptTemplate = SystemMessagePromptTemplate.fromTemplate(
+			`${options.systemPromptTemplate ?? SYSTEM_PROMPT_TEMPLATE}
+{format_instructions}`,
+		);
```

Comment:
We should probably move this inside the for-loop. Because these fields could be expressions that would depend on the individual item, so we should resolve it per-item and not just fallback to first one(0 index)

Reply:
Ah, good that you mention that; because the `TextClassifier Node` also has this problem then!

Reply:
You're right. I missed it when reviewing it :/